savant/gstreamer/metadata.py

```diff
@@ -1,58 +1,67 @@
 """Gstreamer metadata."""
 import logging
-from collections import defaultdict
+from collections import defaultdict, UserDict
 from dataclasses import dataclass, field
+from types import MappingProxyType
 from typing import Any, Dict, Optional, Union
 
 logger = logging.getLogger(__name__)
 
 DEFAULT_FRAMERATE = '30/1'
 
 
+class OnlyExtendedDict(UserDict):
```

Comment:
It's possible for the user to change values in this type of dict by removing keys first via `__delitem__()`, `pop()`, `popitem()`, `clear()`.

Reply:
I have overridden these methods